state-chain/pallets/cf-ingress-egress/src/benchmarking.rs

```diff
@@ -163,6 +169,272 @@ mod benchmarks {
+
+		// Worst-case scenario is when all boosters withdraw funds while
+		// waiting for the deposit to be finalised:
+		for booster_id in &boosters {
+			assert_ok!(Pallet::<T, I>::stop_boosting(
+				RawOrigin::Signed(booster_id.clone()).into(),
+				asset,
+				FEE_TIER
+			));
+		}
+
+		let boost_id = PrewitnessedDepositIdCounter::<T, I>::get();
+
+		#[block]
+		{
+			BoostPools::<T, I>::mutate(asset, FEE_TIER, |pool| {
+				// This depends on the number of boosters who contributed to it:
+				pool.as_mut().unwrap().on_lost_deposit(boost_id);
+			});
+		}
+	}
+
+	#[benchmark]
+	fn stop_boosting() {
+		const FEE_TIER: BoostPoolTier = BoostPoolTier::FiveBps;
+		let asset = TargetChainAsset::<T, I>::iter().next().unwrap();
+
+		let lp_account = setup_booster_account::<T, I>(asset, 0);
+
+		assert_ok!(Pallet::<T, I>::add_boost_funds(
+			RawOrigin::Signed(lp_account.clone()).into(),
+			asset,
+			1_000_000u32.into(),
+			FEE_TIER
+		));
+
+		// `stop_boosting` has linear complexity w.r.t. the number of pending boosts,
+		// and this seems like a reasonable estimate:
+		const PENDING_BOOSTS_COUNT: usize = 100;
+
+		for _ in 0..PENDING_BOOSTS_COUNT {
+			prewitness_deposit::<T, I>(&lp_account, asset, FEE_TIER);
+		}
+
+		#[block]
+		{
+			// This depends on the number active boosts:
+			assert_ok!(Pallet::<T, I>::stop_boosting(
+				RawOrigin::Signed(lp_account).into(),
+				asset,
+				FEE_TIER
+			));
+		}
+
+		assert_eq!(
+			BoostPools::<T, I>::get(asset, FEE_TIER).unwrap().get_available_amount(),
+			0u32.into()
+		);
+	}
+
+	#[benchmark]
+	fn deposit_boosted() {
+		const FEE_TIER: BoostPoolTier = BoostPoolTier::FiveBps;
+		let asset = TargetChainAsset::<T, I>::iter().next().unwrap();
+
+		const BOOSTER_COUNT: u32 = 100;
+
+		let boosters: Vec<_> =
+			(0..BOOSTER_COUNT).map(|i| setup_booster_account::<T, I>(asset, i)).collect();
+
+		for booster_id in &boosters {
+			assert_ok!(Pallet::<T, I>::add_boost_funds(
+				RawOrigin::Signed(booster_id.clone()).into(),
+				asset,
+				1_000_000u32.into(),
+				FEE_TIER
+			));
+		}
+
+		let (_channel_id, deposit_address, ..) = Pallet::<T, I>::open_channel(
+			&boosters[0],
+			asset,
+			ChannelAction::LiquidityProvision { lp_account: boosters[0].clone() },
+			FEE_TIER as u16,
+		)
+		.unwrap();
+
+		let amount_before =
+			BoostPools::<T, I>::get(asset, FEE_TIER).unwrap().get_available_amount();
+
+		#[block]
+		{
+			assert_ok!(Pallet::<T, I>::add_prewitnessed_deposits(
+				vec![DepositWitness::<T::TargetChain> {
+					deposit_address,
+					asset,
+					amount: TargetChainAmount::<T, I>::from(1000u32),
+					deposit_details: BenchmarkValue::benchmark_value()
+				}],
+				BenchmarkValue::benchmark_value()
+			),);
+		}
+
+		// This would fail if the deposit didn't get boosted:
+		assert!(
+			BoostPools::<T, I>::get(asset, FEE_TIER).unwrap().get_available_amount() <
+				amount_before
+		)
+	}
+
+	#[benchmark]
+	fn boost_finalised() {
```

Comment:
just "deposit_finalised"? We use it on process_deposits, and it's not necessarily the case we have a boost

Reply:
Are you asking why this doesn't also contain code for processing deposit normally? Or just about renaming it? I use `boost_finalised` as benchmark for `deposit_finalised` because it is (by far) the more expensive code path. (Is there a better technique for benchmarking code with branches?) I use `boost_finalised` as the name to better reflect the code that's actually being benchmarked.

Reply:
was just suggesting renaming it. usually the benchmarks are named after their extrinsic, I can see this case is a little unique though, we're benchmarking the boost pathway through this extrinsic - so guess I'm indifferent will leave it to you